### PR TITLE
feat(ai-monitoring): Cohere integration

### PR DESCRIPTION
**Pull request by colin-sentry**

### CODE REVIEW

File: sentry_sdk/integrations/cohere.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+            ):
+                if isinstance(kwargs["texts"], str):
+                    set_data_normalized(span, "ai.texts", [kwargs["texts"]])
+                elif (
+                    isinstance(kwargs["texts"], list)
+                    and len(kwargs["texts"]) > 0
+                    and isinstance(kwargs["texts"][0], str)
+                ):
+                    set_data_normalized(
+                        span, SPANDATA.AI_INPUT_MESSAGES, kwargs["texts"]
```
Comment (antonpirker) [inline sentry_sdk/integrations/cohere.py:235]: did you mean `kwargs["texts"][0]`?

File: tests/integrations/cohere/test_cohere.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+                                },
+                            },
+                        }
+                    ),
+                ]
+            ),
+        )
+    )
+
+    with start_transaction(name="openai tx"):
```
Comment (antonpirker) [inline tests/integrations/cohere/test_cohere.py:112]:
```suggestion
    with start_transaction(name="cohere tx"):
```

File: tests/integrations/cohere/test_cohere.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+                "meta": {
+                    "billed_units": {
+                        "input_tokens": 10,
+                    }
+                },
+            },
+        )
+    )
+
+    with start_transaction(name="openai tx"):
```
Comment (antonpirker) [inline tests/integrations/cohere/test_cohere.py:187]:
```suggestion
    with start_transaction(name="cohere tx"):
```

File: tests/integrations/cohere/test_cohere.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+        responses = list(
+            client.chat_stream(
+                model="some-model",
+                chat_history=[ChatMessage(role="SYSTEM", message="some context")],
+                message="hello",
+            )
+        )
+        response_string = responses[-1].response.text
```
Comment (antonpirker) [inline tests/integrations/cohere/test_cohere.py:120]: only the last response is important?

**Review by antonpirker — APPROVED** [review]
Looks good. Just some minor questions and nit picking.

File: sentry_sdk/integrations/cohere.py (hunk 1 above)
Comment (colin-sentry) [reply sentry_sdk/integrations/cohere.py:235]: No, it gets auto-normalized if there is only one

File: tests/integrations/cohere/test_cohere.py (hunk 4 above)
Comment (colin-sentry) [reply tests/integrations/cohere/test_cohere.py:120]: cohere sends a summary response at the end of the stream that contains the entire message content.

**Pull request merged**